CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
Oh also ghc-mod makes sure to only run a single query against cabal-helper to keep the time needed to parse `dist/setup-config` to a minimum but also has the side effect of making this a non-problem.